packages/contracts-bedrock/scripts/autogen/calc-weekly-block/main.go

```diff
@@ -0,0 +1,379 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+)
+
+// Reference points for initial block estimation.
+// These provide a starting point for binary search.
```

Comment:
Can we delete all this new go code and just use `cast find-block` to binary search for us? It should be just a few lines of bash to get the timestamp and pass it to cast